modules/javafx.graphics/src/main/java/com/sun/glass/ui/Application.java

```diff
@@ -768,4 +753,33 @@ public final Optional<Boolean> isKeyLocked(int keyCode) {
                 return Optional.empty();
         }
     }
+
+    /**
+     * Returns the current set of platform properties as a map of platform-specific keys to
+     * arbitrary values.
+     */
+    public Map<String, Object> getPlatformPreferences() {
+        return Map.of();
```

Comment:
Should javadoc explicitly proclaim that the map is immutable?

Reply:
No. This map is only used by `QuantumToolkit` to initialize the preferences. The implementation should not keep this map around, and an immutability guarantee might be seen as an invitation to do so.

Reply:
would it make sense to add this explanation to javadoc, even though it's not a public API?

Reply:
I've added the following sentence:
```
Callers should assume that the returned map is immutable, while implementations should
either return an immutable map, or give up ownership of the returned map.
```

What do you think?

Reply:
Thank you for clarification!